src/net/tcp/listener.rs

```diff
@@ -41,6 +44,20 @@ pub struct TcpListener {
 }
 
 impl TcpListener {
+    /// Creates a new `TcpListener` from a standard `net::TcpListener`.
+    ///
+    /// This function is intended to be used to wrap a TCP listener from the
+    /// standard library in the Mio equivalent. The conversion assumes nothing
+    /// about the underlying listener.
+    pub fn from_std(listener: net::TcpListener) -> Self {
```

Comment:
Can you use `TcpListener` instead of `Self`. I personally use `Self` in build patterns to indicate it returns the same object, but other people might use it differently.

Reply:
Same for the other types.